File: tests/skillmeta.py
#!/usr/bin/env python3
# Validate that a SKILL.md has a LEADING, CLOSED YAML frontmatter block whose
# `name` equals the expected value and whose `description` is non-empty.
# (Greps over the whole file would pass metadata outside the frontmatter — the
# packaging-review blocker this guards against.) Exit 0 iff valid.
import sys

path, expected = sys.argv[1], sys.argv[2]
try:
    lines = open(path).read().split("\n")
except FileNotFoundError:
    print(f"missing: {path}"); sys.exit(1)

if not lines or lines[0].strip() != "---":
    print(f"no leading frontmatter: {path}"); sys.exit(1)

end = next((i for i in range(1, len(lines)) if lines[i].strip() == "---"), None)
if end is None:
    print(f"unclosed frontmatter: {path}"); sys.exit(1)

fm = lines[1:end]

def field(key):
    for l in fm:
        if l.startswith(key + ":"):
            return l[len(key) + 1:].strip()
    return None

name, desc = field("name"), field("description")
if name != expected:
    print(f"name '{name}' != '{expected}': {path}"); sys.exit(1)
if not desc:
    print(f"empty/missing description: {path}"); sys.exit(1)
sys.exit(0)

File: tests/skillmeta.rb
#!/usr/bin/env ruby
# Validate a SKILL.md's LEADING frontmatter with a real YAML parse (codex review: a line/grep
# "parser" accepted unloadable YAML and description: ""). Exit 0 iff: the file opens with a closed
# `--- … ---` block that parses as a YAML mapping whose `name` == expected and whose `description`
# is a non-empty string.
require 'yaml'

path, expected = ARGV[0], ARGV[1]
unless path && expected && File.exist?(path)
  warn "missing or bad args: #{path.inspect}"; exit 1
end

lines = File.readlines(path, chomp: true)
if lines.empty? || lines[0].strip != "---"
  warn "no leading frontmatter: #{path}"; exit 1
end
close = (1...lines.length).find { |i| lines[i].strip == "---" }
if close.nil?
  warn "unclosed frontmatter: #{path}"; exit 1
end

begin
  meta = YAML.safe_load(lines[1...close].join("\n"))
rescue => e
  warn "invalid frontmatter YAML: #{path} (#{e.class})"; exit 1
end
unless meta.is_a?(Hash)
  warn "frontmatter is not a mapping: #{path}"; exit 1
end
if meta["name"] != expected
  warn "name #{meta['name'].inspect} != #{expected.inspect}: #{path}"; exit 1
end
desc = meta["description"]
unless desc.is_a?(String) && !desc.strip.empty?
  warn "missing/empty description: #{path}"; exit 1
end
exit 0
